[15.0][ADD] partner_delivery_link

dsolanki-initos opened the pull request.

pedrobaeza committed:
crm_deduplicate_by_website

===============================
Deduplicate Contacts by website
===============================

This module extends the criteria to match duplicated contacts using the field
website.

Usage
=====

To use this module, you need to:

* Go to *Sales > Tools > Deduplicate Contacts*.
* Mark "Website" in the section "Search duplicates based on duplicated data
  in".
* This criteria will be used for deduplicating.

cubells committed:
[9.0][MIG] crm_deduplicate_by_website module
- Updated README
- Updated views
- Improved tests

Commit:
[MIG] crm_deduplicate_by_website

sbejaoui committed:
[ADD] - Add partner_industry_parent module

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (2 of 2 strings)

Translation: crm-11.0/crm-11.0-crm_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/crm-11-0/crm-11-0-crm_deduplicate_by_website/gl/

Commit:
[MIG] crm_deduplicate_by_website > partner_deduplicate_by_website: Migration to v12

Commit:
[FIX] Rename translation files: crm_deduplicate_by_website > partner_deduplicate_by_website

sbejaoui committed:
[FIX] - Fix gnu link

sbejaoui committed:
[IMP] - Add unit tests

sbejaoui committed:
[IMP] - Add complete name to the form view

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_deduplicate_by_website/pt_BR/

kevinkhao committed:
[IMP] partner_deduplicate_by_website: black, isort

kevinkhao committed:
[MIG] partner_deduplicate_by_website: Migration to 13.0

oca-travis committed:
[UPD] Update partner_industry_parent.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
partner_industry_parent 12.0.2.0.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_deduplicate_by_website/es_AR/

Rad0van committed:
Fix website attribute in manifest

mariadforgeflow committed:
[MIG] partner_deduplicate_by_website: Migration to 14.0

chienandalu committed:
[ADD] partner_pricelist_tracking: New module

TT35444

dsolanki-initos committed:
[15.0][ADD] partner_delivery_link

dsolanki-initos committed:
[15.0][ADD] partner_delivery_link

fshah-initos reviewed (approved):
Code review and functional tests.
Looks good to me.

rousseldenis commented:
@dsolanki-initos Thanks for this. Is this really a migration ?

dsolanki-initos commented:
> @dsolanki-initos Thanks for this. Is this really a migration ?

No it is new module for this repository

dsolanki-initos commented:
@CarlosRoca13 Can you please review this PR?

dsolanki-initos commented:
@ajaniszewska-dev Can you please review this PR?

ajaniszewska-dev commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,36 @@
+
+    delivery_count = fields.Integer(compute="_compute_delivery_count")
+
+    def _compute_delivery_count(self):
+        delivery_obj = self.env["stock.picking"]
+        for partner in self:
+            partner.delivery_count = delivery_obj.search_count(
+                [
+                    ("partner_id", "=", partner.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
+
+    def action_view_partner_delivery(self):
+        delivery_orders = self.env["stock.picking"].search(
+            [
+                ("partner_id", "=", self.id),
+                ("picking_type_id.code", "=", "outgoing"),
+            ]
+        )
+        delivery_tree_view = self.env.ref("stock.vpicktree")
+        delivery_form_view = self.env.ref("stock.view_picking_form")
```
for this, domain in both methods is same, so maybe it is worth to consider to have fields like:
M2X delivery_order_ids and integer delivery_count, use one computed method (_compute_delivery_count) for both fields and then:
- we search by this domain inside compute method and assign records  to delivery_order_ids
- delivery_count would be just len(self.delivery_count_ids). 

With this approach we are sure that both fields which rely on same recordset are computed in one method.
Then in domain returned by action we refer to delivery_order_ids.ids

dsolanki-initos replied:
Thank you for reviewing the code and I will modify the code as per your suggesion

ajaniszewska-dev reviewed (approved):
LG, minor comment

ikapasi-initos reviewed (approved):
Code Review and functional test.
LGTM

CarlosRoca13 commented:
Change the commit message to `[ADD] partner_delivery_link: Little description of module` please

dsolanki-initos commented:
> Change the commit message to `[ADD] partner_delivery_link: Little description of module` please

Okay I will add description

CarlosRoca13 commented:
> > Change the commit message to `[ADD] partner_delivery_link: Little description of module` please
> 
> Okay I will add description

It's more important set the correct tag. This is not a [MIG], but an [ADD]

dsolanki-initos commented:
@CarlosRoca13 @ajaniszewska-dev  Here is the new PR https://github.com/OCA/partner-contact/pull/1305 Please review it

dsolanki-initos commented:
@leemannd Could you please review this PR?

ajaniszewska-dev commented on partner_delivery_link/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,2 @@
+Partner Delivery Link module is a part of partner-contact and count the totals picking
+created for customer in customer form and can access all pickings.
```
```suggestion
Partner Delivery Link module counts the total pickings
created for customer and adds button (on customer form) to access them.
```

something like that

ajaniszewska-dev reviewed (approved):
minor comment, LG

dsolanki-initos commented:
Can we merge this PR?

victoralmau committed:
[ADD] partner_country_lang: New addon.
TT38559

victoralmau committed:
[DCK] Pre-commit fixes

AungKoKoLin1997 commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,34 @@
+
+    def _compute_delivery_count(self):
+        for partner in self:
+            delivery_order_ids = self.env["stock.picking"].search(
+                [
+                    ("partner_id", "=", self.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
+            partner.delivery_order_ids = delivery_order_ids
+            partner.delivery_count = len(delivery_order_ids)
```
Can you use the function of this picking search? So, It can make picking search flexible.

AungKoKoLin1997 commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,34 @@
+
+    def _compute_delivery_count(self):
+        for partner in self:
+            delivery_order_ids = self.env["stock.picking"].search(
+                [
+                    ("partner_id", "=", self.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
```
Can you add the function for picking search? So, It can make picking search flexible.

AungKoKoLin1997 commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,34 @@
+
+    def _compute_delivery_count(self):
+        for partner in self:
+            delivery_order_ids = self.env["stock.picking"].search(
+                [
+                    ("partner_id", "=", self.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
```
```suggestion
            delivery_order_ids = self._get_partner_pickings(partner)
```

AungKoKoLin1997 replied:
Can I please suggest you to make picking search easier to manipulate with the domain condition? If users use this module , they can add more domain as per their requirement easily. So, when we use function for this picking search, I think that is more clear code. As an example, If module user also want to show child pickings and his pickings together, they need to add new domain like ['|',("partner_id", "in", self.child_ids.ids),("partner_id", "=", self.id),  ("picking_type_id.code", "=", "outgoing")] and they can adapt easily.

AungKoKoLin1997 commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,34 @@
+        "stock.picking", compute="_compute_delivery_count"
+    )
+    delivery_count = fields.Integer(compute="_compute_delivery_count")
+
```
```suggestion  

    @api.model
    def _domain_for_pickings_search(self, partner):
        return [
                    ("partner_id", "=", partner.id),
                    ("picking_type_id.code", "=", "outgoing"),
                ]
                
    @api.model
    def _get_partner_pickings(self, partner):
        domain = self._domain_for_pickings_search(partner)
        return self.env['stock.picking'].search(domain)
        
```

AungKoKoLin1997 commented on partner_delivery_link/models/res_partner.py:
```diff
@@ -0,0 +1,34 @@
+from odoo import fields, models
```
```suggestion
from odoo import fields, models, api
```

dsolanki-initos replied:
@AungKoKoLin1997 I've applied changes can you please review it?

klodr committed:
Translated using Weblate (French)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_deduplicate_by_website/fr/

MiquelRForgeFlow committed:
[FIX] partner_company_group: avoid non-deterministic error

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 50.0% (1 of 2 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_deduplicate_by_website/it/

stefan-tecnativa committed:
[IMP] partner_pricelist_tracking: black, isort, prettier

stefan-tecnativa committed:
[MIG] partner_pricelist_tracking: Migration to 15.0

eantones committed:
[FIX] base_location: City/locations completion object display_name is not updated when changing the city name

Commit:
[UPD] Update partner_pricelist_tracking.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_country_lang
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_country_lang/

ernesto-garcia-tecnativa committed:
[IMP] partner_country_lang: black, isort, prettier

ernesto-garcia-tecnativa committed:
[MIG] partner_country_lang: Migration to 15.0

Commit:
[UPD] Update partner_country_lang.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
sale_partner_company_group 15.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

jans23 commented:
I see this PR has several reviews already. Could anybody merge it or at least apply the "no stale" label, please?

NL66278 reviewed (changes requested):
Please follow the migration method that preserves commit history outlined here:  https://helpdesk.therp.nl/web#id=589&action=2406&model=document.page&view_type=form&cids=1&menu_id=1631

jans23 commented:
@NL66278 This is not a migration but a new module. AFAIK in this case it should contain exactly one commit, right?

BTW, the link you shared is not accessible without credentials.

NL66278 commented:
@jans23 I was wrong-footed by the `mig` part in your branch name. Sorry for that.

NL66278 reviewed (approved):
:+1: After a fresh look LGTM

NL66278 commented:
/ocabot merge nobump

ernesto-garcia-tecnativa committed:
[IMP] partner_deduplicate_by_website: black, isort, prettier

ernesto-garcia-tecnativa committed:
[MIG] partner_deduplicate_by_website: Migration to 15.0

Commit:
[UPD] Update partner_deduplicate_by_website.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-15.0/partner-contact-15.0-partner_deduplicate_by_website
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-partner_deduplicate_by_website/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 80.5% (29 of 36 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_location/it/

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

yankinmax committed:
[MIG] partner_industry_parent: Migration to 15.0

OCA-git-bot committed:
base_location 15.0.1.0.2

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[UPD] Update partner_industry_parent.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 53.3% (8 of 15 strings)

Translation: partner-contact-15.0/partner-contact-15.0-partner_employee_quantity
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-partner_employee_quantity/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-15.0/partner-contact-15.0-partner_contact_job_position
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-partner_contact_job_position/it/

Commit:
[15.0][ADD] test case added and CI checks fixed

rousseldenis commented:
@dsolanki-initos @alihashaam Could you squash, rebase and fix the conflict ?

Commit:
Merge branch '15.0' into 15.0-mig-partner_delivery_link

Commit:
[15.0][ADD] dependency for delivery module added

Commit:
[15.0][ADD] dependency for delivery module added

Commit:
[15.0][FIX] test cases changed regarding the scope of the module

Commit:
[15.0][FIX] test cases changed regarding the scope of the module

alihashaam commented:
@rousseldenis 

Sure, thanks. I will do that

Commit:
[MIG][ADD] conflicts removed in readme.md

Commit:
Merge branch '15.0-mig-partner_delivery_link' of github.com:Nitrokey/odoo-partner-contact into 15.0-mig-partner_delivery_link

Commit:
Merge branch '15.0' of github.com:Nitrokey/odoo-partner-contact into 15.0-mig-partner_delivery_link

Commit:
[15.0][fix] base_location and partner_ref_unique: test company named changed in tests

Commit:
[15.0][fix] base_location and partner_ref_unique: test company name changes are reverted

This reverts commit bf5de6c26296d46b4c269a7efdc22182bf4a1086.

alihashaam commented:
@rousseldenis There are issues in CI because of the tests in modules "base_location" and "partner_ref_unique" which are basically not part of the PR. Should they be resolved after rebasing? any pointers?

NL66278 commented:
/ocabot rebase

NL66278 commented:
@dsolanki-initos The module has merge conflicts that basically having nothing to do with the module you want to add. My suggestion, rename your branch, recreate a new branch with the old name (15.0-mig-partner_delivery_link). Copy your module from the renamed branch to the new branch, commit (so the whole module is one commit) and force push to here.

Alternatively close this PR and resubmit with a branch name 15.0-add-partner_delivery_link, making clear this is a new module, not a migration.

NL66278 commented:
Work taken up in new PR #1600
